src/compiler/crystal/syntax/lexer.cr

```diff
@@ -3078,6 +3076,9 @@ module Crystal
       if error = @reader.error
         ::raise InvalidByteSequenceError.new("Unexpected byte 0x#{error.to_s(16)} at position #{@reader.pos}, malformed UTF-8")
       end
+      if current_char.in?('\u202A', '\u202B', '\u202C', '\u202D', '\u202E', '\u2066', '\u2067', '\u2068', '\u2069')
```

Comment:
How do we handle nonbreaking space? (U+00A0) That also seems potentially confusing.

Reply:
Yes, non-breaking space and other invisible characters can be confusing. But at the worst they're just invisible by themselves. They don't manipulate the display of of other visible characters like bidi control characters.
Of course, they should be invalid in most places as well, but I wouldn't necessarily go as far as disallow them entirely. That could be an option, but this definitely needs more discussion to happen in #11216.